### ISSUE TITLE
Start scrapyd server error

### ISSUE THREAD

**Issue by lx1054331851** [scrapy/scrapyd#370]
I install anaconda, scrapy, scrapyd on my ubuntu18.04, when i try to start scrapyd server, below error accured:
```python

(base) lee@ScrapyStation:~$ scrapyd
Unhandled Error
Traceback (most recent call last):
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/application/app.py", line 674, in run
    runApp(config)
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/scripts/twistd.py", line 25, in runApp
    runner.run()
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/application/app.py", line 381, in run
    self.application = self.createOrGetApplication()
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/application/app.py", line 453, in createOrGetApplication
    application = getApplication(self.config, passphrase)
--- <exception caught here> ---
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/application/app.py", line 464, in getApplication
    application = service.loadApplication(filename, style, passphrase)
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/application/service.py", line 416, in loadApplication
    application = sob.loadValueFromFile(filename, 'application')
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/persisted/sob.py", line 177, in loadValueFromFile
    eval(codeObj, d, d)
  File "/home/lee/anaconda3/lib/python3.7/site-packages/scrapyd/txapp.py", line 3, in <module>
    application = get_application()
  File "/home/lee/anaconda3/lib/python3.7/site-packages/scrapyd/__init__.py", line 14, in get_application
    appfunc = load_object(apppath)
  File "/home/lee/anaconda3/lib/python3.7/site-packages/scrapy/utils/misc.py", line 50, in load_object
    mod = import_module(module)
  File "/home/lee/anaconda3/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
    
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
    
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
    
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
    
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
    
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
    
  File "/home/lee/anaconda3/lib/python3.7/site-packages/scrapyd/app.py", line 2, in <module>
    from twisted.application.internet import TimerService, TCPServer
  File "/home/lee/anaconda3/lib/python3.7/site-packages/twisted/application/internet.py", line 54, in <module>
    from automat import MethodicalMachine
  File "/home/lee/anaconda3/lib/python3.7/site-packages/automat/__init__.py", line 2, in <module>
    from ._methodical import MethodicalMachine
  File "/home/lee/anaconda3/lib/python3.7/site-packages/automat/_methodical.py", line 209, in <module>
    @attr.s(eq=False, hash=False)
builtins.TypeError: attrs() got an unexpected keyword argument 'eq'


Failed to load application: attrs() got an unexpected keyword argument 'eq'

```

**Comment by my8100** [comment]
Can you post the result of command “pip list”?
For comparison, can you new a virtual env and install Scrapyd solely?

**Comment by Lofoer** [comment]
Scrapy                1.8.0
scrapy-redis          0.6.8
scrapyd               1.2.1
scrapyd-client        1.1.0
Twisted               19.10.0
I'm will create new env try again.

<img width="959" alt="Xnip2020-03-13_19-24-13" src="https://user-images.githubusercontent.com/10608420/76617139-9a681680-6560-11ea-894b-8515e3ecc61b.png">

**Comment by Lofoer** [comment]
now, I create new_env run scrapyd is ok. So Cool.

<img width="1005" alt="Xnip2020-03-13_22-10-14" src="https://user-images.githubusercontent.com/10608420/76628280-72d07880-6577-11ea-9e49-655761626f00.png">

**Comment by lx1054331851** [comment]
pip list result is :
```python
(base) lee@ScrapyStation:~$ pip list
Package                               Version
------------------------------------- -------------------
adal                                  1.2.2
alabaster                             0.7.12
anaconda-client                       1.7.2
anaconda-navigator                    1.9.12
anaconda-project                      0.8.3
APScheduler                           3.6.3
asn1crypto                            1.3.0
astroid                               2.3.3
astropy                               4.0
atomicwrites                          1.3.0
attrs                                 19.3.0
Automat                               20.2.0
azure                                 4.0.0
azure-applicationinsights             0.1.0
azure-batch                           4.1.3
azure-common                          1.1.25
azure-cosmosdb-nspkg                  2.0.2
azure-cosmosdb-table                  1.0.6
azure-datalake-store                  0.0.48
azure-eventgrid                       1.3.0
azure-graphrbac                       0.40.0
azure-keyvault                        1.1.0
azure-loganalytics                    0.1.0
azure-mgmt                            4.0.0
azure-mgmt-advisor                    1.0.1
azure-mgmt-applicationinsights        0.1.1
azure-mgmt-authorization              0.50.0
azure-mgmt-batch                      5.0.1
azure-mgmt-batchai                    2.0.0
azure-mgmt-billing                    0.2.0
azure-mgmt-cdn                        3.1.0
azure-mgmt-cognitiveservices          3.0.0
azure-mgmt-commerce                   1.0.1
azure-mgmt-compute                    4.6.2
azure-mgmt-consumption                2.0.0
azure-mgmt-containerinstance          1.5.0
azure-mgmt-containerregistry          2.8.0
azure-mgmt-containerservice           4.4.0
azure-mgmt-cosmosdb                   0.4.1
azure-mgmt-datafactory                0.6.0
azure-mgmt-datalake-analytics         0.6.0
azure-mgmt-datalake-nspkg             3.0.1
azure-mgmt-datalake-store             0.5.0
azure-mgmt-datamigration              1.0.0
azure-mgmt-devspaces                  0.1.0
azure-mgmt-devtestlabs                2.2.0
azure-mgmt-dns                        2.1.0
azure-mgmt-eventgrid                  1.0.0
azure-mgmt-eventhub                   2.6.0
azure-mgmt-hanaonazure                0.1.1
azure-mgmt-iotcentral                 0.1.0
azure-mgmt-iothub                     0.5.0
azure-mgmt-iothubprovisioningservices 0.2.0
azure-mgmt-keyvault                   1.1.0
azure-mgmt-loganalytics               0.2.0
azure-mgmt-logic                      3.0.0
azure-mgmt-machinelearningcompute     0.4.1
azure-mgmt-managementgroups           0.1.0
azure-mgmt-managementpartner          0.1.1
azure-mgmt-maps                       0.1.0
azure-mgmt-marketplaceordering        0.1.0
azure-mgmt-media                      1.0.0
azure-mgmt-monitor                    0.5.2
azure-mgmt-msi                        0.2.0
azure-mgmt-network                    2.7.0
azure-mgmt-notificationhubs           2.1.0
azure-mgmt-nspkg                      3.0.2
azure-mgmt-policyinsights             0.1.0
azure-mgmt-powerbiembedded            2.0.0
azure-mgmt-rdbms                      1.9.0
azure-mgmt-recoveryservices           0.3.0
azure-mgmt-recoveryservicesbackup     0.3.0
azure-mgmt-redis                      5.0.0
azure-mgmt-relay                      0.1.0
azure-mgmt-reservations               0.2.1
azure-mgmt-resource                   2.2.0
azure-mgmt-scheduler                  2.0.0
azure-mgmt-search                     2.1.0
azure-mgmt-servicebus                 0.5.3
azure-mgmt-servicefabric              0.2.0
azure-mgmt-signalr                    0.1.1
azure-mgmt-sql                        0.9.1
azure-mgmt-storage                    2.0.0
azure-mgmt-subscription               0.2.0
azure-mgmt-trafficmanager             0.50.0
azure-mgmt-web                        0.35.0
azure-nspkg                           3.0.2
azure-servicebus                      0.21.1
azure-servicefabric                   6.3.0.0
azure-servicemanagement-legacy        0.20.6
azure-storage-blob                    1.5.0
azure-storage-common                  1.4.2
azure-storage-file                    1.4.0
azure-storage-queue                   1.4.0
Babel                                 2.8.0
backcall                              0.1.0
backports.functools-lru-cache         1.6.1
backports.os                          0.1.1
backports.shutil-get-terminal-size    1.0.0
backports.tempfile                    1.0
backports.weakref                     1.0.post1
beautifulsoup4                        4.8.2
bitarray                              1.2.1
bkcharts                              0.2
bleach                                3.1.0
bokeh                                 2.0.0
boto                                  2.49.0
Bottleneck                            1.3.2
certifi                               2019.11.28
cffi                                  1.14.0
chardet                               3.0.4
Click                                 7.0
cloudpickle                           1.3.0
clyent                                1.2.2
colorama                              0.4.3
conda                                 4.8.2
conda-build                           3.18.11
conda-package-handling                1.6.0
conda-verify                          3.4.2
constantly                            15.1.0
contextlib2                           0.6.0.post1
cryptography                          2.8
cssselect                             1.1.0
cycler                                0.10.0
Cython                                0.29.15
cytoolz                               0.10.1
dask                                  2.12.0
decorator                             4.4.1
defusedxml                            0.6.0
dictmysql                             0.6.8
distributed                           2.12.0
docutils                              0.16
entrypoints                           0.3
et-xmlfile                            1.0.1
fastcache                             1.1.0
filelock                              3.0.12
Flask                                 1.1.1
Flask-Compress                        1.4.0
Flask-SQLAlchemy                      2.4.1
fsspec                                0.6.2
future                                0.18.2
gevent                                1.4.0
glob2                                 0.7
gmpy2                                 2.0.8
greenlet                              0.4.15
h5py                                  2.10.0
HeapDict                              1.0.1
html5lib                              1.0.1
hyperlink                             19.0.0
hypothesis                            5.5.4
idna                                  2.8
imageio                               2.6.1
imagesize                             1.2.0
importlib-metadata                    1.5.0
incremental                           17.5.0
ipykernel                             5.1.4
ipython                               7.13.0
ipython-genutils                      0.2.0
ipywidgets                            7.5.1
isodate                               0.6.0
isort                                 4.3.21
itsdangerous                          1.1.0
jdcal                                 1.4.1
jedi                                  0.16.0
jeepney                               0.4.2
Jinja2                                2.11.1
joblib                                0.14.1
Js2Py                                 0.68
json5                                 0.9.1
jsonschema                            3.2.0
jupyter                               1.0.0
jupyter-client                        5.3.4
jupyter-console                       6.1.0
jupyter-core                          4.6.1
jupyterlab                            1.2.6
jupyterlab-server                     1.0.6
keyring                               21.1.0
kiwisolver                            1.1.0
lazy-object-proxy                     1.4.3
libarchive-c                          2.8
lief                                  0.9.0
llvmlite                              0.31.0
locket                                0.2.0
logparser                             0.8.2
lxml                                  4.5.0
MarkupSafe                            1.1.1
matplotlib                            3.1.3
mccabe                                0.6.1
mistune                               0.8.4
mkl-fft                               1.0.15
mkl-random                            1.1.0
mkl-service                           2.3.0
mock                                  4.0.1
more-itertools                        8.2.0
mpmath                                1.1.0
msgpack                               0.6.1
msrest                                0.6.11
msrestazure                           0.6.2
multipledispatch                      0.6.0
navigator-updater                     0.2.1
nbconvert                             5.6.1
nbformat                              5.0.4
networkx                              2.4
nltk                                  3.4.5
nose                                  1.3.7
notebook                              6.0.3
numba                                 0.48.0
numexpr                               2.7.1
numpy                                 1.18.1
numpydoc                              0.9.2
oauthlib                              3.1.0
olefile                               0.46
openpyxl                              3.0.3
packaging                             20.1
pandas                                1.0.1
pandocfilters                         1.4.2
parsel                                1.5.2
parso                                 0.6.1
partd                                 1.1.0
path                                  13.1.0
pathlib2                              2.3.5
patsy                                 0.5.1
pep8                                  1.7.1
pexpect                               4.8.0
pickleshare                           0.7.5
Pillow                                7.0.0
pip                                   20.0.2
pkginfo                               1.5.0.1
pluggy                                0.13.1
ply                                   3.11
prometheus-client                     0.7.1
prompt-toolkit                        3.0.3
Protego                               0.1.16
psutil                                5.6.7
ptyprocess                            0.6.0
py                                    1.8.1
pyasn1                                0.4.8
pyasn1-modules                        0.2.8
pycodestyle                           2.5.0
pycosat                               0.6.3
pycparser                             2.19
pycrypto                              2.6.1
pycurl                                7.43.0.5
PyDispatcher                          2.0.5
pyflakes                              2.1.1
Pygments                              2.5.2
PyHamcrest                            2.0.2
pyjsparser                            2.7.1
PyJWT                                 1.7.1
pylint                                2.4.4
PyMySQL                               0.9.3
pyodbc                                4.0.0-unsupported
pyOpenSSL                             19.1.0
pyparsing                             2.4.6
pyrsistent                            0.15.7
PySocks                               1.7.1
pytest                                5.0.1
pytest-arraydiff                      0.3
pytest-astropy                        0.8.0
pytest-astropy-header                 0.1.2
pytest-doctestplus                    0.5.0
pytest-openfiles                      0.4.0
pytest-remotedata                     0.3.2
python-dateutil                       2.8.1
pytz                                  2019.3
PyWavelets                            1.1.1
PyYAML                                5.3
pyzmq                                 18.1.1
QtAwesome                             0.6.1
qtconsole                             4.7.1
QtPy                                  1.9.0
queuelib                              1.5.0
redis                                 3.4.1
requests                              2.23.0
requests-oauthlib                     1.3.0
rope                                  0.16.0
ruamel-yaml                           0.15.87
scikit-image                          0.16.2
scikit-learn                          0.22.1
scipy                                 1.4.1
Scrapy                                2.0.0
scrapyd                               1.2.1
scrapydweb                            1.4.0
seaborn                               0.10.0
SecretStorage                         3.1.2
Send2Trash                            1.5.0
service-identity                      18.1.0
setuptools                            46.0.0.post20200309
simplegeneric                         0.8.1
singledispatch                        3.4.0.3
six                                   1.14.0
snowballstemmer                       2.0.0
sortedcollections                     1.1.2
sortedcontainers                      2.1.0
soupsieve                             1.9.5
Sphinx                                2.4.0
sphinxcontrib-applehelp               1.0.1
sphinxcontrib-devhelp                 1.0.1
sphinxcontrib-htmlhelp                1.0.2
sphinxcontrib-jsmath                  1.0.1
sphinxcontrib-qthelp                  1.0.2
sphinxcontrib-serializinghtml         1.1.3
sphinxcontrib-websupport              1.2.0
spyder                                3.3.6
spyder-kernels                        0.5.2
SQLAlchemy                            1.3.13
statsmodels                           0.11.0
sympy                                 1.5.1
tables                                3.6.1
tblib                                 1.6.0
terminado                             0.8.3
testpath                              0.4.4
toolz                                 0.10.0
tornado                               6.0.3
tqdm                                  4.42.1
traitlets                             4.3.3
Twisted                               19.10.0
typing-extensions                     3.7.4.1
tzlocal                               2.0.0
unicodecsv                            0.14.1
urllib3                               1.25.8
w3lib                                 1.21.0
wcwidth                               0.1.8
webencodings                          0.5.1
Werkzeug                              1.0.0
wheel                                 0.34.2
widgetsnbextension                    3.5.1
wrapt                                 1.11.2
wurlitzer                             2.0.0
xlrd                                  1.2.0
XlsxWriter                            1.2.7
xlwt                                  1.3.0
xmltodict                             0.12.0
zict                                  1.0.0
zipp                                  2.2.0
zope.interface                        4.7.2

```

**Comment by lx1054331851** [comment]
After I create a new env, scrapyd start successfully.
```python
(scrapyer) lee@ScrapyStation:~$ pip list
Package          Version
---------------- -------------------
attrs            19.3.0
Automat          20.2.0
certifi          2019.11.28
cffi             1.14.0
constantly       15.1.0
cryptography     2.8
cssselect        1.1.0
hyperlink        19.0.0
idna             2.9
incremental      17.5.0
lxml             4.5.0
parsel           1.5.2
pip              20.0.2
Protego          0.1.16
pyasn1           0.4.8
pyasn1-modules   0.2.8
pycparser        2.20
PyDispatcher     2.0.5
PyHamcrest       2.0.2
pyOpenSSL        19.1.0
queuelib         1.5.0
Scrapy           2.0.0
scrapyd          1.2.1
service-identity 18.1.0
setuptools       46.0.0.post20200309
six              1.14.0
Twisted          19.10.0
w3lib            1.21.0
wheel            0.34.2
zope.interface   4.7.2
```

```python
(scrapyer) lee@ScrapyStation:~$ scrapyd
2020-03-14T12:13:53+0000 [-] Loading /home/lee/anaconda3/envs/scrapyer/lib/python3.7/site-packages/scrapyd/txapp.py...
2020-03-14T12:13:53+0000 [-] Scrapyd web console available at http://127.0.0.1:6800/
2020-03-14T12:13:53+0000 [-] Loaded.
2020-03-14T12:13:53+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] twistd 19.10.0 (/home/lee/anaconda3/envs/scrapyer/bin/python 3.7.6) starting up.
2020-03-14T12:13:53+0000 [twisted.scripts._twistd_unix.UnixAppLogger#info] reactor class: twisted.internet.epollreactor.EPollReactor.
2020-03-14T12:13:53+0000 [-] Site starting on 6800
2020-03-14T12:13:53+0000 [twisted.web.server.Site#info] Starting factory <twisted.web.server.Site object at 0x7f25398eb150>
2020-03-14T12:13:53+0000 [Launcher] Scrapyd 1.2.1 started: max_proc=8, runner='scrapyd.runner'
^C2020-03-14T12:13:58+0000 [-] Received SIGINT, shutting down.
2020-03-14T12:13:58+0000 [-] (TCP Port 6800 Closed)
2020-03-14T12:13:58+0000 [twisted.web.server.Site#info] Stopping factory <twisted.web.server.Site object at 0x7f25398eb150>
```

**Comment by lx1054331851** [comment]
Today I update conda use "conda update --all", scrapyd start successfully!
